### PR TITLE
Add contains and ends-with Metapath functions

### PR DESCRIPTION
**Pull request by david-waltermire**
# Committer Notes

Added support for the contains and ends-with Metapath functions.

Also, cleaned up signatures for existing tests to reduce their visability.

### All Submissions:

- [x] Have you followed the guidelines in our [Contributing](https://github.com/metaschema-framework/metaschema-java/blob/main/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/metaschema-framework/metaschema-java/pulls) for the same update/change?
- [x] Have you squashed any non-relevant commits and commit messages? \[[instructions](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History)\]
- [ ] Have you set "[Allow edits and access to secrets by maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork)"?

By submitting a pull request, you are agreeing to provide this contribution under the [CC0 1.0 Universal public domain](https://creativecommons.org/publicdomain/zero/1.0/) dedication.

### Changes to Core Features:

- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your core changes, as applicable?
- [ ] Have you included examples of how to use your new feature(s)?
- [ ] Have you updated all website and readme documentation affected by the changes you made?

### CODE REVIEW

File: core/src/main/java/gov/nist/secauto/metaschema/core/metapath/function/library/FnEndsWith.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+      IItem focus) {
+    IStringItem arg1 = FunctionUtils.asTypeOrNull(arguments.get(0).getFirstItem(true));
+
+    IStringItem arg2 = FunctionUtils.asTypeOrNull(arguments.get(1).getFirstItem(true));
+
+    return ISequence.of(IBooleanItem.valueOf(
+        fnEndsWith(
+            arg1 == null ? "" : arg1.asString(),
+            arg2 == null ? "" : arg2.asString())));
+  }
```
Comment (aj-stein-gsa) [inline core/src/main/java/gov/nist/secauto/metaschema/core/metapath/function/library/FnEndsWith.java:59]: Non-blocking, but important: per [the documentation](https://www.w3.org/TR/xpath-functions-31/#func-ends-with), we are not going to implement the three argument form? I am asking, I know that the collation handling may be juice not worth the squeeze, but you get me reading official docs a lot now, so I thought I may as well as ask.

**Review by aj-stein-gsa — APPROVED** [review]
Nice work, your speed gives me anxiety. 😅 

1. Slightly more code, but I really like the `... NAME = "...";` constants to make copy-pasting a relevant function given similar args and behavior even easier for a novice to use and dig in with easily tactical replacements and even easier to spot diffs on changes and/or new function definition. It makes me happy.
2. I do not want to block, but I noticed on the two new fns in question we are not implementing the three arg variant altogether. I am not saying I think that is important, but we may want to document that we do not support (for now or semi-permanent WONTFIX emphasis) and explain why. But that is very minor consideration not worthy of blocking a PR.

Again, nice work!

File: core/src/main/java/gov/nist/secauto/metaschema/core/metapath/function/library/FnContains.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+      IItem focus) {
+    IStringItem arg1 = FunctionUtils.asTypeOrNull(arguments.get(0).getFirstItem(true));
+
+    IStringItem arg2 = FunctionUtils.asTypeOrNull(arguments.get(1).getFirstItem(true));
+
+    return ISequence.of(IBooleanItem.valueOf(
+        fnContains(
+            arg1 == null ? "" : arg1.asString(),
+            arg2 == null ? "" : arg2.asString())));
+  }
```
[the remark this answers is not in the document]
Comment (david-waltermire) [reply core/src/main/java/gov/nist/secauto/metaschema/core/metapath/function/library/FnContains.java:59]: I can add an issue to add support for collations and the related function variants.

[the remark this answers is not in the document]
Comment (david-waltermire) [reply core/src/main/java/gov/nist/secauto/metaschema/core/metapath/function/library/FnContains.java:59]: See https://github.com/metaschema-framework/metaschema-java/issues/170

File: core/src/main/java/gov/nist/secauto/metaschema/core/metapath/function/library/FnEndsWith.java (hunk 1 above)
Comment (david-waltermire) [reply core/src/main/java/gov/nist/secauto/metaschema/core/metapath/function/library/FnEndsWith.java:59]: See #170

**Review by aj-stein-gsa — APPROVED** [review]

**Comment by aj-stein-gsa** [comment]
Sorry, I didn't catch the merge conflict before ...

**Pull request merged**